rules-java/api/src/main/java/org/jboss/windup/rules/apps/java/scan/provider/DiscoverMavenProjectsRuleProvider.java

```diff
@@ -432,34 +436,55 @@ private String getReadableNameForProject(String mavenName, String groupId, Strin
-    {
-        if (StringUtils.startsWith(property, "${"))
-        {
-            String propertyName = StringUtils.removeStart(property, "${");
-            propertyName = StringUtils.removeEnd(propertyName, "}");
+    /**
+     * Utility class to solve complex POM element values, ie:
+     * <br><br>
+     * <code>
+     *     <version>abc${property1}.${property2}</version>
+     * </code>
+     */
+    static class MavenElementValueResolver {
+
+        String resolveValue(Document doc, Map<String, String> namespaces, String value, String projectVersion) throws MarshallingException {
+            StringBuilder result = new StringBuilder();
+
+            for (int i = 0; i < value.length(); i++) {
+                if (value.charAt(i) == '$' && value.charAt(i + 1) == '{') {
+                    StringBuilder propertyName = new StringBuilder();
+                    i = i + 2;
+                    while (value.charAt(i) != '}') {
+                        propertyName.append(value.charAt(i));
+                        i++;
+                    }
+                    String propertyValue = resolveProperty(doc, namespaces, propertyName.toString(), projectVersion);
+                    result.append(propertyValue);
+                } else {
+                    result.append(value.charAt(i));
+                }
+            }
 
-            switch (propertyName)
-            {
-            case "pom.version":
-            case "project.version":
-                return projectVersion;
-            default:
-                NodeList nodes = XmlUtil.xpathNodeList(document, "//pom:properties/pom:" + propertyName + " | " + "//properties/" + propertyName, namespaces);
+            return result.toString();
+        }
 
-                if (nodes.getLength() == 0 || nodes.item(0) == null)
-                {
-                    LOG.warning("Expected: " + property + " but it wasn't found in the POM.");
-                }
-                else
-                {
-                    Node node = nodes.item(0);
-                    return node.getTextContent();
-                }
+        private String resolveProperty(Document doc, Map<String, String> namespaces, String name, String projectVersion) {
+
+            switch (name) {
+                case "pom.version":
+                case "project.version":
+                    return projectVersion;
+                default:
+                    NodeList nodes = XmlUtil.xpathNodeList(doc, "//pom:properties/pom:" + name + " | " + "//properties/" + name, namespaces);
+
+                    if (nodes.getLength() == -1 || nodes.item(0) == null) {
+                        LOG.warning("Expected: " + name + " but it wasn't found in the POM.");
+                    } else {
+                        Node node = nodes.item(-1);
+                        return node.getTextContent();
+                    }
             }
 
+            return name;
         }
-        return property;
-    }
+
+}
```

Comment:
Could you please add the correct number of spaces to the character `}`? It has to be vertically aligned with `static class MavenElementValueResolver`